### ISSUE TITLE
ONIX-Schema.org crosswalk clarifications needed

### ISSUE THREAD

**Issue by rickj** [w3c/publ-a11y#263]
Questions about the Schema.org Accessibility Properties Crosswalk document at https://w3c.github.io/a11y-discov-vocab/crosswalk/
(referencing the Draft Community Group Report 09 February 2024)

In the EPUB and ONIX section
- For 'certifierReport', there are two items listed.  Is this an OR, a SHOULD, or a MUST?

In the Schema.org and ONIX section
- For 'latex' the qualifier 'should be used with' is difficult (impossible?) to create code around.  What are the consequences of making this a MUST?
- For accessMode/auditory it specifies Code 01:Audiobook plus types of audio content.  Is this an OR, SHOULD, or MUST?  Also, specify where we find the list of 'plus types of audio content'

General questions
- ONIX Code 09, Code 10, Code 22, and Code 24 are not mapped to anything in EPUB schema.  Can you confirm that we can ignore these?
- The following EPUB Schema elements are not mapped to anything in ONIX.  Can you confirm that we can ignore: annotations, bookmarks (deprecated), captions, certifiedBy, colorDependent, fullRubyAnnotations, horizontalWriting, none, pageNavigation, rubyAnnotations, signLanguage, tactile, tactileGraphic, tactileObject, textual, timingControl, transcript, unknown, unknownFlashingHazard, unknownMotionSimulationHazard, unknownSoundHazard, verticalWriting, visual, withAdditionalWordSegmentation, withoutAdditionalWordSegmentation

**Comment by chrisONIX** [comment]
The following EPUB Schema elements are not mapped to anything in ONIX. Can you confirm that we can ignore: annotations, bookmarks (deprecated), captions, certifiedBy, colorDependent, fullRubyAnnotations, horizontalWriting, none, pageNavigation, rubyAnnotations, signLanguage, tactile, tactileGraphic, tactileObject, textual, timingControl, transcript, unknown, unknownFlashingHazard, unknownMotionSimulationHazard, unknownSoundHazard, verticalWriting, visual, withAdditionalWordSegmentation, withoutAdditionalWordSegmentation

I know that some of these have equivalents and I have been working through them - I can do an updated list. I also know that some of these we have outstanding questions about if they are being used by anyone in the book supply chain yet.

**Comment by chrisONIX** [comment]
or accessMode/auditory it specifies Code 01:Audiobook plus types of audio content. Is this an OR, SHOULD, or MUST? Also, specify where we find the list of 'plus types of audio content'

I would say this should be an OR 

The list is part of ONIX code list 81

**Comment by chrisONIX** [comment]
For 'certifierReport', there are two items listed. Is this an OR, a SHOULD, or a MUST?

I would suggest this should be an OR

**Comment by amesga** [comment]
> The following EPUB Schema elements are not mapped to anything in ONIX. Can you confirm that we can ignore: annotations, bookmarks (deprecated), captions, certifiedBy, colorDependent, fullRubyAnnotations, horizontalWriting, none, pageNavigation, rubyAnnotations, signLanguage, tactile, tactileGraphic, tactileObject, textual, timingControl, transcript, unknown, unknownFlashingHazard, unknownMotionSimulationHazard, unknownSoundHazard, verticalWriting, visual, withAdditionalWordSegmentation, withoutAdditionalWordSegmentation
> 
> I know that some of these have equivalents and I have been working through them - I can do an updated list. I also know that some of these we have outstanding questions about if they are being used by anyone in the book supply chain yet. Do you want me to list the suggested updates in this thread?

This would be helpful for us, thank you. This is the list we have been referencing: https://w3c.github.io/a11y-discov-vocab/crosswalk/

**Comment by chrisONIX** [comment]
annotations - we suggest using list 21 - code ANN - https://ns.editeur.org/onix/en/21/ANN 
/ONIXMessage/​Product[]/​DescriptiveDetail​[EditionType eq 'ANN' ]

bookmarks (deprecated), - for legacy data for PDF bookmarks we suggest using code 05 from list 196 - PDF/UA - https://ns.editeur.org/onix/en/196/05
/ONIXMessage/​Product[]/​DescriptiveDetail/​ProductFormFeature [ProductFormFeatureType  eq '09']  [ProductFormFeatureValue  eq '05'] 

captions ( (deprecated),  - ONIX has Closed captions (List 175 -V210) and Open Captions (list 175 - V211) - so for legacy data you would have to map to one of these two.
https://ns.editeur.org/onix/en/175/V210
/ONIXMessage/​Product[]/​DescriptiveDetail​[ProductFormDetail eq 'V210' ]
https://ns.editeur.org/onix/en/175/V211
/ONIXMessage/​Product[]/​DescriptiveDetail​[ProductFormDetail eq 'V211' ]

certifiedBy - we recommend using list 196 code 93 - https://ns.editeur.org/onix/en/196/93 - as an other / or with certifierCredential

colorDependent, - ONIX has the opposite - a positive declaration - Use of color is not sole means of conveying information - from list 196- code 25 
/ONIXMessage/​Product[]/​DescriptiveDetail/​ProductFormFeature [ProductFormFeatureType  eq '09']  [ProductFormFeatureValue  eq '25'] 

fullRubyAnnotations, - we have a reserved code but waiting feedback if being used in the supply chain already
horizontalWriting, -  we have a reserved code but waiting feedback if being used in the supply chain already

none - ONIX code list 143 - code 00 - https://ns.editeur.org/onix/en/143/00 
/ONIXMessage/​Product[]/​DescriptiveDetail/​ProductFormFeature [ProductFormFeatureType  eq '12']  [ProductFormFeatureValue  eq '00'] 

pageNavigation - after discussion we recommend using code list 196 - code 19 - /ONIXMessage/​Product[]/​DescriptiveDetail/​ProductFormFeature [ProductFormFeatureType  eq '09']  [ProductFormFeatureValue  eq '19'] 

rubyAnnotations - -  we have a reserved code but waiting feedback if being used in the supply chain already

signLanguage - in ONIX the use of sign language is sent like this : 
<Language>	
    <LanguageRole>09</LanguageRole> 	
    <LanguageCode>sgn</LanguageCode> 
    <CountryCode>US</CountryCode> 
</Language>
We have a reserved code for list 175 - to indicate the product has sign language - but again this depends on need in supply chain.


tactile, tactileGraphic, tactileObject,  - all these have reserved codes but we are waiting feedback on if they are being used in the supply chain yet.

timingControl - we have suggested list 196 - code 31 - https://ns.editeur.org/onix/en/196/31 
/ONIXMessage/​Product[]/​DescriptiveDetail/​ProductFormFeature [ProductFormFeatureType  eq '09']  [ProductFormFeatureValue  eq '31'] 
(Although this is one we would happy to discuss more)

transcript - in ONIX the transcript is sent using list 158 code 47 - Transcript (Full transcript of audio or video content of the product) - this is a link to the actual transcript.
We also have a reserved code in list 175 - to say the product has a transcript - but we are waiting to know if this would be useful addition or not.

unknown, - ONIX code list 196 - code 08 - https://ns.editeur.org/onix/en/196/08 
/ONIXMessage/​Product[]/​DescriptiveDetail/​ProductFormFeature [ProductFormFeatureType  eq '09']  [ProductFormFeatureValue  eq '08'] 

 unknownFlashingHazard, unknownMotionSimulationHazard, unknownSoundHazard, - these are not clear to us and we wanted to know what they meant - is this unknown but present or not known if there is one? 

visual - this is expressed in ONIX using one or more codes from list 81 - these codes indicate visual - 07, 18, 19, 20, 12, 46, 42, 50, 24, 44 
https://ns.editeur.org/onix/en/81


verticalWriting,  withAdditionalWordSegmentation, withoutAdditionalWordSegmentation - all these have reserved codes but we are waiting feedback on if they are being used in the supply chain yet.


I hope this helps. It would be good to discuss what can be added to the crosswalk and also where we have reserved codes - if there is any current use of these values in the supply chain.

**Comment by clapierre** [comment]
> unknownFlashingHazard, unknownMotionSimulationHazard, unknownSoundHazard, - these are not clear to us and we wanted to know what they meant - is this unknown but present or not known if there is one?

This means not know if these hazards exist.
Eg. Publisher has an audio clip in their EPUB.  They know for sure there is noMotionSimulationHazard, and noFlashingHazard but has no idea if there is any sound hazard associated with the audio clip so the would in this case put "unknownSoundHazard".

**Comment by amesga** [comment]
Thanks everyone, I've added your notes to our internal documentation. Just to summarize (and ask a few more questions, naturally):

Still waiting on feedback for the following codes
- accessMode:tactile
- fullRubyAnnotations
- horizontalWriting
- rubyAnnotations
- tactileGraphic
- tactileObject
- verticalWriting
- withAdditionalWordSegmentation
- withoutAdditionalWordSegmentation

My new questions:

- Are there any concerns with us making the "should be used with" for the LaTeX calculation into a "must"?
- The calculation for Audiobook says List 81:01 "plus types of audio content". Where should we expect to see these types?
- Per your suggestions, 196:93 could map to either certifiedBy or certifierCredential. Would you recommend storing the same information in both fields, or should we only fill out one of the fields?
- unknownFlashingHazard, unknownMotionSimulationHazard, unknownSoundHazard, are all under the accessibilityHazard schema, but have no ONIX value in the crosswalk. Is it not possible to model these values?
- Onix codelist 196 values 9, 10, 22, 24, do not map to any accessibilityFeatures in the crosswalk. I'd just like to confirm that there is in fact no mapping to these fields.

**Comment by gregoriopellegrino** [comment]
@chrisONIX may you check this issue, if it is resolved with the new version of crosswalk?

**Comment by chrisONIX** [comment]
Onix codelist 196 values 9, 10, 22, 24, do not map to any accessibilityFeatures in the crosswalk. I'd just like to confirm that there is in fact no mapping to these fields.

Code 09 has been mapped to none 
Code 10 has no direct mapping.
Code 22 has no direct mapping
Code 24 has no direct mapping

unknownFlashingHazard, unknownMotionSimulationHazard, unknownSoundHazard, are all under the accessibilityHazard schema, but have no ONIX value in the crosswalk. Is it not possible to model these values?

I will open a new issue with the unmapped values in hazards
